code/game/objects/effects/glowshroom.dm

```diff
@@ -37,17 +40,42 @@
 			return FALSE
 	. = ..()
 
-/obj/structure/glowshroom/Crossed(AM as mob|obj)
-	if(isliving(AM))
-		var/mob/living/L = AM
-		if(L.z == z)
-			if(!HAS_TRAIT(L, TRAIT_KNEESTINGER_IMMUNITY))
-				if(L.electrocute_act(30, src))
-					L.emote("painscream")
-					L.update_sneak_invis(TRUE)
-					L.consider_ambush(always = TRUE)
+/obj/structure/glowshroom/proc/can_zap(atom/movable/movable_victim)
+	if(!isliving(movable_victim))
+		return FALSE
+	var/mob/living/victim = movable_victim
+	if(HAS_TRAIT(victim, TRAIT_KNEESTINGER_IMMUNITY)) //Dendor kneestinger immunity
+		return FALSE
+	if(victim.throwing)	//Exemption from floor hazard, you're thrown over it.
+		victim.throwing.finalize(FALSE)
+	//if(victim.is_floor_hazard_immune)	//Floating, flying, etc
```

Comment:
this can probably be removed, it existed mostly to clean up seelie code. it also factored in throwing though